### ISSUE TITLE
Attiny84 Keyboard

### ISSUE THREAD

**Issue by Soumojit28** [parmaja/usbKeyboard#1]
Can you help me how can I use this with attiny84 to act as a  keyboard?

**Comment by zaher** [comment]
follow this project it is based on another one with improvements, it is IR project but you can modify it to send keyboards

https://github.com/parmaja/usbIRConsumer

**Comment by Soumojit28** [comment]
Thanks for the quick answer, I have some more questions as I am using an attiny84 is it require an external 16Mhz crystal to communicate with the usb?  and need a bit more help with the schematics, which pins on attiny84 should connect to usb?

**Comment by zaher** [comment]
look at this project https://nathan.chantrell.net/20121014/tinypcremote-an-attiny85-based-infrared-pc-remote-control/
you can remove the part of IR reciever, and keep attiny send keyboard message by adding push button for example

**Comment by Soumojit28** [comment]
Yup, I sawed it, but the thing is will this stays the same for an attiny84, and any ideas if I need an external crystal for it?

**Comment by zaher** [comment]
just add it, look at this schema, i used external crustal 20 for my project

https://www.avrfreaks.net/forum/implement-usb-device-attiny85

**Comment by Soumojit28** [comment]
So tried it, added a 16Mhz crystal in attiny84, its working with the crystal, modified the usbconfig.h like this 

``/* This is the port where the USB bus is connected. When you configure it to
 * "B", the registers PORTB, PINB and DDRB will be used.
 */
#define USB_CFG_DMINUS_BIT      1
/* This is the bit number in USB_CFG_IOPORT where the USB D- line is connected.
 * This may be any bit in the port.
 */
#define USB_CFG_DPLUS_BIT       2
/* This is the bit number in USB_CFG_IOPORT where the USB D+ line is connected.
 * This may be any bit in the port. Please note that D+ must also be connected
 * to interrupt pin INT0! [You can also use other interrupts, see section
 * "Optional MCU Description" below, or you can connect D- to the interrupt, as
 * it is required if you use the USB_COUNT_SOF feature. If you use D- for the
 * interrupt, the USB interrupt will also be triggered at Start-Of-Frame
 * markers every millisecond.]
 */
#define USB_CFG_CLOCK_KHZ       16000``


Compiled the example sketch in the library and uploaded the code but getting Unknown USB Device (Device Descriptor Failed) in the device manager. any ideas whats wrong here?

**Comment by zaher** [comment]
I can not help you, because it is not my main job, it is just hobby project.
